doc/casync.rst

```diff
@@ -241,3 +246,46 @@ Archive features
 The various ``--with=`` and ``--without=`` parameters control the precise set
 of metadata to store in the archive, or restore when extracting. These flags
 only apply if ``casync`` operates on the file system level.
+
+Excluding Files and Directories from Archiving
+----------------------------------------------
+
+When generating an archive or index from a file system directory tree, some
+files and directories are excluded by default and others may optionally be
+excluded:
+
+1. Files and directories of virtual API file systems exposed by the kernel
+   (i.e. procfs, sysfs, …) are excluded unconditionally.
+
+2. Depending on whether symlinks, device nodes, fifos and sockets are enabled
+   for archiving with ``--with=`` and ``--without=``, file nodes of these types
+   are excluded.
+
+3. By default, files and directories with the ``+d`` chattr(1) flag set are
+   excluded, however this behaviour may be turned off with
+   ``--exclude-nodump=no``.
+
+4. Optionally, files and directories contained in submounts of the specified
+   file system tree are excluded, if ``--exclude-submounts=yes`` is specified.
+
+5. By default, any files and directories listed in ``.caexclude`` files in the
+   file hierarchy are excluded, however interpretation of these files may be
+   turned off with ``--exclude-file=no``. These files operate similar to
+   ``git``'s ``.gitignore`` concept: they are read as text file where each line
+   is either empty/starts with ``#`` (in which case they have no effect, which
+   may be used for commenting), or list a globbing path pattern of
+   files/directories to ignore. If a line contains no ``/`` character the line
+   applies to the directory the ``.caexclude`` file is located in as well as
+   all child directories of it. If it contains at least one ``/`` character it
+   is considered stricly relative to the directory the ``.caexclude`` file is
+   located in. ``.caexclude`` files may appear in any directory of the file
+   system tree that is archived, however they have no effect when placed in
+   directories that are marked for exclusion via ``.caexclude`` files placed
+   further up in the directory tree. When a line ends in a ``/`` character it
+   applies to directories only, and not regular files or other file node
+   types. If a line is prefixed with a ``!`` character matching files are
+   excluded from the exclusion, i.e. the effect of other matching lines that
+   are not prefixed like this is cancelled for matching files. ``!`` lines
+   unconditionally take precedence over lines not marked like this. Moreover,
+   lines prefixed with ``!`` also cancel the effect of patterns in
+   ``.caexclude`` files placed in directories further up the tree.
```

Comment:
I understand that the idea is to be gitignore-compatible. I wonder if we shouldn't just say that and refer people to gitignore(5) for details?

Reply:
Well, there are slight differences in the semantics, specifically regarding spaces, and the `**` stuff... We might want to add support for that later on, but for now I didn't want to refer to something that is more comprehensive than what we do right now...

Or to say this differently: when we actually reach feature parity we can refer to gitignore directly for the syntax... But until then I think we should document what precisely we do, but not more